### PR TITLE
Improved default value casting in QuantityValues

### PR DESCRIPTION
**Pull request by NiklasBr**
This is a follow-up to #10772 which only partially solved the issue

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/QuantityValue.php [hunk 1]
```diff
@@ -155,7 +155,9 @@ public function setUnitWidth($unitWidth)
      */
     public function getDefaultValue()
     {
-        if (!empty($this->defaultValue)) {
+        // Default values such as "0" and "0.0" should be accepted and returned as float,
+        // while empty strings and nulls should not
+        if (!empty($this->defaultValue) || is_numeric($this->defaultValue)) {
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/QuantityValue.php]:
```suggestion
        if (is_numeric($this->defaultValue)) {
```

File: models/DataObject/ClassDefinition/Data/QuantityValue.php [hunk 2]
```diff
@@ -167,7 +169,7 @@ public function getDefaultValue()
      */
     public function setDefaultValue($defaultValue)
     {
-        if (!empty($defaultValue)) {
+        if (!empty($defaultValue) || is_numeric($defaultValue)) {
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/QuantityValue.php]:
```suggestion
        if (is_numeric($defaultValue)) {
```

**Comment by NiklasBr** [comment]
Makes total sense @blankse!

**Comment by brusch** [comment]
Very nice - thanks a lot @NiklasBr @blankse !

**Pull request merged**